acl/acl.go

```diff
@@ -307,6 +340,42 @@ func (a *ACL) matchingNamespaceCapabilitySet(ns string) (capabilitySet, bool) {
 	return a.findClosestMatchingGlob(a.wildcardNamespaces, ns)
 }
 
+// anyNamespaceAllowsOp returns true if any namespace in ACL object allows the
+// given operation.
+func (a *ACL) anyNamespaceAllowsOp(op string) bool {
+	return a.anyNamespaceAllows(func(c capabilitySet) bool {
+		return c.Check(op)
+	})
+}
+
+// anyNamespaceAllowsAnyOp returns true if any namespace in ACL object allows
+// at least one operation.
+func (a *ACL) anyNamespaceAllowsAnyOp() bool {
+	return a.anyNamespaceAllows(func(c capabilitySet) bool {
+		return len(c) > 0 && !c.Check(PolicyDeny)
+	})
+}
+
+// anyNamespaceAllows returns true if the callback cb returns true for any
+// namespace operation of the ACL object.
+func (a *ACL) anyNamespaceAllows(cb func(capabilitySet) bool) bool {
+	allow := false
+
+	checkFn := func(_ []byte, iv interface{}) bool {
+		v := iv.(capabilitySet)
```

Comment:
do we not need to check this is okay? if we can statically assert, shouldn't the checkFn parameter be a `capabilitySet`?

Reply:
`checkFn` needs to be a [`WalkFn`](https://pkg.go.dev/github.com/hashicorp/go-immutable-radix#WalkFn), so the signature is pretty set.

This iterates over the internal tree of namespaces, if there's data that is not a `capabilitySet` then something pretty wrong happened 😅 

[`findAllMatchingWildcards`](https://github.com/hashicorp/nomad/blob/v1.3.1/acl/acl.go#L361-L363) doesn't have the check either, so I think it's safe to assume this would always work?

Reply:
Ahh I see, yup just ignore me!